lib/statsd.js

```diff
@@ -1,9 +1,42 @@
 "use strict";
 var TXStatsD = require('node-txstatsd');
+var StatsD = require('node-statsd');
 
-// StatsD wrapper
-function StatsD(options) {
-    this.statsd = new TXStatsD({
+var nameCache = {};
+function normalizeName (name) {
+    // See https://github.com/etsy/statsd/issues/110
+    // Only [\w_.-] allowed, with '.' being the hierarchy separator.
+    var res = nameCache[name];
+    if (res) {
+        return res;
+    } else {
+        nameCache[name] = name.replace( /[^\/a-zA-Z0-9\.\-]/g, '-' )
```

Comment:
The intention is to map `_` to `-`, so that slashes are the only char that produces underscores in the final stat name. This is admittedly a bit optimized for restbase path needs.